tf2_sensor_msgs/test/test_tf2_sensor_msgs.py

```diff
@@ -45,7 +45,48 @@ def test_simple_transform(self):
         assert(expected_coordinates == new_points)
         assert(old_data == self.point_cloud_in.data)  # checking no modification in input cloud
 
+
+## A sample python unit test (multi channel version)
+class PointCloudConversionsMultichannel(unittest.TestCase):
+    def setUp(self):
+        self.point_cloud_in = point_cloud2.PointCloud2()
+        self.point_cloud_in.fields = [PointField('x', 0, PointField.FLOAT32, 1),
+                                PointField('y', 4, PointField.FLOAT32, 1),
+                                PointField('z', 8, PointField.FLOAT32, 1),
+                                PointField('index', 12, PointField.INT32, 1)]
+
+        self.point_cloud_in.point_step = 4 * 4
+        self.point_cloud_in.height = 1
+        # we add two points (with x, y, z to the cloud)
+        self.point_cloud_in.width = 2
+        self.point_cloud_in.row_step = self.point_cloud_in.point_step * self.point_cloud_in.width
+
+        points = [1, 2, 0, 123, 10, 20, 30, 456]
+        self.point_cloud_in.data = struct.pack('3fi3fi', *points)
+
+
+        self.transform_translate_xyz_300 = TransformStamped()
+        self.transform_translate_xyz_300.transform.translation.x = 300
```

Comment:
It would be great to reuse k from https://github.com/ros/geometry2/pull/241/files#diff-1eb6abfaf9a31338a0944f59de492214R80 maybe make that a constant at the top `TRANSFORM_OFFSET_DISTANCE = 300`